### PR TITLE
ci: upload production ready extension artifacts as the release assets

### LINKED ISSUE #63: Append production ready zip file of extension in the release tags
### Description

On every github release, we want to have a archive file of dist folder attached to the respective release artifacts.

Screenshot from 2024-07-17 13-23-43

Proposed archive filename: `mdpresentation-{release-tag}.zip` (e.g.: `mdpresentation-1.0.0.zip`)
Also attach `md5` and `sha256` sums.

See https://github.com/owncloud/web-extensions/releases

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Whenever there's a release, upload the necessary artifacts to the release assets. These assets will be added:
- `mdpresentation-{release_tag}.zip`
- `md5sum.txt`
- `sha256sum.txt`

## Related Issue
- Fixes #63

## Screenshots (if appropriate):

Screenshot from 2024-07-17 17-10-44

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [ ] Documentation only (no source changes)
- [ ] Maintenance (e.g. dependency updates or tooling)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation updated

### CODE REVIEW

**Review by S-Panta — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,6 +33,14 @@ jobs:
       - name: build
         run: pnpm build
 
+      - name: upload-dist
+        uses: actions/upload-artifact@v4
+        if: github.event_name == 'release'
+        with:
+          name: dist
+          path: dist/
+          retention-days: 1
```
Comment (S-Panta) [inline .github/workflows/ci.yml:42]: Is there any reason to choose 1 day for artifact retention time?

Comment (saw-jan) [reply .github/workflows/ci.yml:42]: we don't need it once it is added to release assets

Comment (saw-jan) [reply .github/workflows/ci.yml:42]: this artifact and release assets are different things

**Pull request merged**